addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -948,13 +948,19 @@ export class Rtc extends Record {
                 const sendCamera = force ?? !this.state.sendCamera;
                 this.state.sendCamera = false;
                 await this.setVideo(track, type, sendCamera);
+                if (!this.state.sourceCameraStream) {
+                    return;
+                }
                 break;
             }
             case "screen": {
                 const track = this.state.screenTrack;
                 const sendScreen = force ?? !this.state.sendScreen;
                 this.state.sendScreen = false;
                 await this.setVideo(track, type, sendScreen);
+                if (!this.state.sourceScreenStream) {
+                    return;
+                }
                 break;
```

Comment:
This doesn't solve the current issue. But, it is preventive measure. So that we don't end up with bugs which we are not aware about. And it also doesn't make sense to execute further if you don't have the necessary access.

Reply:
I prefer no diff, more LOCs require more maintenance, especially when they are hard to trace motivation. Besides this makes harder to remove / change `sourceScreeStream` state because we are scared to change these LOCs.

Reply:
Also code of discuss calls is decent, but there's still ongoing code improvements and code cleaning planned eventually. We notably plan to use very soon Discuss calls for internal uses, so we'll most likely be most sensitive to making the code more robust, with better code architecture without extra hard-to-maintain LOC like these ones.